Web Serial API - consistency updates (#44153)
mdn/content: files/en-us/web/api/serialport/close/index.md

However, when continuously reading data from a serial device using a loop, the associated [readable stream](/en-US/docs/Web/API/ReadableStream) will always be locked until the [reader](/en-US/docs/Web/API/ReadableStreamDefaultReader) encounters an error. In this case, calling [`reader.cancel()`](/en-US/docs/Web/API/ReadableStreamDefaultReader/cancel) will force [`reader.read()`](/en-US/docs/Web/API/ReadableStreamDefaultReader/read) to resolve immediately with `{ value: undefined, done: true }` allowing the loop to call [`reader.releaseLock()`](/en-US/docs/Web/API/ReadableStreamDefaultReader/releaseLock).

Closing a serial port is more complicated when using [transform streams](/en-US/docs/Web/API/TransformStream). See [Close a serial port](https://developer.chrome.com/docs/capabilities/serial#close-port) for guidance.

## Examples

> ### Closing a port after reading until the stream is done

Reviewer: Sounds a bit confused. How about 

```suggestion
### Closing a port after reading once the stream is done
```

> The following example shows how to close a port while continuously reading data from it. A `keepReading` flag is used to control when to stop reading. A button click sets `keepReading` to `false` and cancels the reader, which causes `reader.read()` to resolve immediately so the loop can release the lock and `close()` can be called.

Reviewer: ```suggestion
The following example shows how to close a port after continuously reading data from it, once the stream is done.
A `keepReading` flag controls when to stop reading.
A button click sets `keepReading` to `false` and cancels the reader, which causes `reader.read()` to resolve immediately so the loop can release the lock and `close()` can be called.
```
